### ISSUE TITLE
Environment variables aren't exposing to the browser

### ISSUE THREAD

**Issue by sinnrrr** [blitz-js/blitz#2108]
### What is the problem?

Using the official guide [here](https://blitzjs.com/docs/environment-variables#exposing-environment-variables-to-the-browser) tried to expose environment variables to the browser

### Paste all your error logs here:

No error logs

### Paste all relevant code snippets here:

`.env.local`
```
NEXT_PUBLIC_ANALYTICS_ID=abcdefghijk
```

`index.tsx`
```
process.env.NEXT_PUBLIC_ANALYTICS_ID // undefined
```

### What are detailed steps to reproduce this?

A new project with env variable `NEXT_PUBLIC_ANALYTICS_ID` and `console.log(process.env.NEXT_PUBLIC_ANALYTICS_ID)` on `index.tsx` `Home()` method

### Run `blitz -v` and paste the output here:

```
Linux 5.8 | linux-x64 | Node: v15.11.0

blitz: 0.32.1 (global)
blitz: 0.32.1 (local)

  Package manager: yarn 
  System:
    OS: Linux 5.8 Pop!_OS 20.04 LTS
    CPU: (8) x64 Intel(R) Xeon(R) CPU E3-1535M v5 @ 2.90GHz
    Memory: 48.28 GB / 62.49 GB
    Shell: 5.8 - /usr/bin/zsh
  Binaries:
    Node: 15.11.0 - /usr/bin/node
    Yarn: 1.22.10 - /usr/bin/yarn
    npm: 7.6.3 - /usr/bin/npm
    Watchman: Not Found
  npmPackages:
    @prisma/client: ~2.17 => 2.17.0 
    blitz: 0.32.1 => 0.32.1 
    prisma: ~2.17 => 2.17.0 
    react: 0.0.0-experimental-3310209d0 => 0.0.0-experimental-3310209d0 
    react-dom: 0.0.0-experimental-3310209d0 => 0.0.0-experimental-3310209d0 
    typescript: ~4.1 => 4.1.5 
```

**Comment by flybayer** [comment]
Hmm... Do you have `.env` in `.gitignore` by chance?

**Comment by Roesh** [comment]
Wait, I thought the process.env.VARIABLE set was only available on the backend?

Also, @sinnrrr to clarify, you mean index.tsx correct? I assume that's what you meant because only that file has a `const Home: BlitzPage` component.

I would assume that this is expected behavior because we're meant to store sensitive things like the database connection strings etc in .env, and we don't want users to be able to console.log(process.env.NEXT_PUBLIC_ANALYTICS_ID) from the frontend

I might be missing something here though, is there is a way to make .env variables appear on the frontend apart from writing a query that simply returns the value from process.env.NEXT_PUBLIC_ANALYTICS_ID from the backend?

**Comment by sinnrrr** [comment]
> Hmm... Do you have `.env` in `.gitignore` by chance?

No, `.env` isn't gitignored

I have made a mistake in the issue. This environment variable is being set not in `.env`, but in `.env.local`, but, according to the documentation, it shouldn't be a problem, because variables are being loaded from `.env*`

**Comment by sinnrrr** [comment]
Also, I have tried both prefixing variable with `NEXT_PUBLIC_`, and setting it in `env` section of `blitz.config.js`, but that didn't work either

**Comment by olerichter00** [comment]
Hey @sinnrrr,

I can't reproduce the bug and all env variables prefixed with `NEXT_PUBLIC_` are accessible in the frontend. 

Did you restart blitz after changing `.env.local`?

**Comment by flybayer** [comment]
I think this is a bug in our current compiler setup that will be fixed once we integrate the next.js fork.

For now, try this — it should work.

```ts
// .env.local
PUBLIC_ANALYTICS_ID=abcdefghijk
```

```ts
// blitz.config.js
module.exports = {
  env: {
    PUBLIC_ANALYTICS_ID: process.env.PUBLIC_ANALYTICS_ID,
  }
}
```

Then you can use `process.env.PUBLIC_ANALYTICS_ID` in client-side code

**Comment by sinnrrr** [comment]
@flybayer @olerichter00 and others

thanks a lot for your time
the issue _just gone_

prefixing with `NEXT_PUBLIC_` is working as expected
I think that might be because I didn't restarted the blitz server

@flybayer thanks for a great framework and great community

**Comment by Roesh** [comment]
I ran into this same issue when I was trying this feature out for the first time. Even though I added the NEXT_PUBLIC_ thing to the variable, I wasn't seeing it pop up on the front end. 

I ran blitz build and blitz start and it worked. I then re-ran blitz dev, and that also worked. I think the reason for this is because

`This inlining occurs at build time, so your various NEXT_PUBLIC_ envs need to be set when the project is built.` as mentioned in the [nextjs doc](https://nextjs.org/docs/basic-features/environment-variables)
I will open a PR to update the docs to clarify this a bit more later so that users know to re-run blitz dev to pickup any newly added environment variables
